### PR TITLE
[KB-17161] Automated release for npm packages

### PR DESCRIPTION
**Pull request by xripoll-at-wiris**
## Description

This PR adds a new GitHub Actions workflow capable of automatically publishing the front-end npm packages.

It has two main additions:

- ADRs, justifying the technical decisions taken. Please read those and comment if there is any disagreements.
- Code changes, including:
  - A new workflow (`.github/workflows/publish.yml`).
  - A new Expect script (`scripts/publish.exp`) in charge of managing Lerna, used by the previous workflow.

---

[#taskid 17161](https://wiris.kanbanize.com/ctrl_board/2/cards/17161/details/)

### CODE REVIEW

**Review by icaparros-at-wiris — COMMENTED** [review]
Awesome ADRs.

Few days ago, I merged the ADR `005-Use-HTML-sanitizer-to-avoid-XSS-attacks.md` on the `KB-15953-BUGSPRINT` branch. We should rename this ADR to 006-Use-DOMPurify...
or rename your both ADR prefix to 006 and 007.

What It makes more sense?

File: resources/publish/htpasswd [hunk 1]
```diff
@@ -0,0 +1 @@
+username:$apr1$gwQzvrYX$hLkvErWORju3C1RS2.QrW0
```
Comment (carla-at-wiris) [inline resources/publish/htpasswd:1]: is the password for the user name stored as private in some place?

File: docs/adr/006-use-expect-for-automating-tui-interaction.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+Shell scripting is a very powerful, but dangerous, solution.
+
+- Good, because it doesn't any dependencies.
+- Bad, because it is very hard to write robust scripts.
+- Bad, because it is complicated to interact with the output of the TUI to automate.
+
+#### Expect
+
+CNPM stands for "Company npm" and is a private npm registry software.
```
Comment (dcanet-at-wiris) [inline docs/adr/006-use-expect-for-automating-tui-interaction.md:41]: Typo alert.
I guess the right text for this paragraph would be:
> "_Expect is a tool for automating interactive applications such as telnet, ftp, passwd, fsck, rlogin, tip, etc._"
> From [Expect website](https://core.tcl-lang.org/expect/index)

**Review by dcanet-at-wiris — CHANGES_REQUESTED** [review]

**Comment by dcanet-at-wiris** [comment]
I've added "Expect Language Support" as a recommended extension to the project.
Since we are adding this extension to our current tool set, I think it's a good idea to add it here, too, for other contributors.

File: docs/adr/006-use-expect-for-automating-tui-interaction.md (hunk 2 above)
Comment (dcanet-at-wiris) [reply docs/adr/006-use-expect-for-automating-tui-interaction.md]: I fixed myself, @xripoll-at-wiris

**Comment by dcanet-at-wiris** [comment]
> Awesome ADRs.
> 
> Few days ago, I merged the ADR `005-Use-HTML-sanitizer-to-avoid-XSS-attacks.md` on the `KB-15953-BUGSPRINT` branch. We should rename this ADR to 006-Use-DOMPurify... or rename your both ADR prefix to 006 and 007.
> 
> What It makes more sense?

Thanks for mentioning it @icaparros-at-wiris.

In order to help things out, I've renamed the Verdaccio's ADR myself to number 007, so we avoid the coincidence with #479.

Thanks!

**Comment by dcanet-at-wiris** [comment]
**Important update**.

This PR's branch has been rebased with the `KB-15953-BUGSPRINT` branch.

Update you local branches, accordingly, please.

Sorry for any incovenience.

**Review by dcanet-at-wiris — COMMENTED** [review]
## Adding `.npmrc` file to the repository 🙀 

I have a question regarding the `.npmrc` file. I am not quite sure if adding this file to the repo is a good move.

If I am not wrong, the default decission would be to include this file to `.gitignore` so it's not added to the repository at all.

I understand that the Verdaccio configuration needs to be included on the repository, in order to work, though.

I am reading the [Verdaccio's documentation site](https://verdaccio.org/docs/authentication/) and I guess one could argue that the user, (developer), could follow some instructions themselves and/or automate the `.npmrc` creation somehow.

```bash
# Add the registry key:
npm adduser --registry http://localhost:4873
# More stuff...
```

What's your take?

Or does the fact that it's not stored on the root folder, and it's more a boilerplate file for an effective CI/CD workflow configuration, that made you decide this option?

File: resources/publish/htpasswd (hunk 1 above)
Comment (dcanet-at-wiris) [reply resources/publish/htpasswd:1]: I'd be more interested to know which command you've used to generate the username token for "username". 😄 

Something like:

```bash
echo -n 'username' | openssl base64
```

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+# Array of jobs to run in this workflow
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      # - name: Merge master to stable
```
[the remark this answers is not in the document]
Comment (xripoll-at-wiris) [reply .github/workflows/publish.yml:38]: No, I forgot to remove them. Thanks.

File: resources/publish/htpasswd (hunk 1 above)
Comment (xripoll-at-wiris) [reply resources/publish/htpasswd:1]: Good note, I'll add a comment.

> is the password for the user name stored as private in some place?

This line is a pair of the form `${username}:${hashed password}`.
It is unsafe to upload your personal `.htpasswd`, but this username:password combination is just literally `username:password`. This file is used to authenticate in the local Verdaccio instance, it's not used in an actual server or in production.

> I'd be more interested to know which command you've used to generate the username token for "username". smile

Also very good note. The command is:

```sh
htpasswd -c htpasswd username
```

Which prompts for a password. Note that the output will be (probably) be different every time you generate this file, even if you enter the same password.

**Comment by xripoll-at-wiris** [comment]
> Or does the fact that it's not stored on the root folder, and it's more a boilerplate file for an effective CI/CD workflow configuration, that made you decide this option?

Effectively this. The intention is not to override user settings, just to use this file when called on purpose. This shouldn't affect anything else. I think it's wise to store configurations outside of code, e.g. in this case in a separate config file.

**Review by dcanet-at-wiris — APPROVED** [review]

**Pull request merged**